ambry-store/src/main/java/com/github/ambry/store/StoreMetrics.java

```diff
@@ -285,11 +298,30 @@ private void deregisterHardDeleteMetric(String storeId) {
     registry.remove(MetricRegistry.name(PersistentIndex.class, prefix + "HardDeleteCaughtUp"));
   }
 
-  void initializeCompactorGauges(String storeId, final AtomicBoolean compactionInProgress) {
+  void initializeCompactorGauges(String storeId, final AtomicBoolean compactionInProgress,
+      AtomicReference<CompactionDetails> compactionDetailsAtomicReference) {
     String prefix = storeId + SEPARATOR;
     Gauge<Long> compactionInProgressGauge = () -> compactionInProgress.get() ? 1L : 0L;
     registry.register(MetricRegistry.name(BlobStoreCompactor.class, prefix + "CompactionInProgress"),
         compactionInProgressGauge);
+    Gauge<Long> compactionCost = () -> {
```

Comment:
I have a big concern about this,  we might have several compactions running at various disks and since they all using the same StoreMetric object, they will override each other.
And since `MetricRegistry.register` is not an idempotent method, if the first compaction registered these metric, the others would throw an exception. we should probably use `MetricRegitry.gauge` method to register this.

Reply:
The metric's name has a prefix, which is storeId, which should be good. No?

Reply:
yeah, this is good.